GSA-TTS/FAC
# Added logic to validate finding reference uniqueness

## Description
 This improves validation for finding references in the audit findings workbook (See #3948 for more details).

**How to Test**:
1. Launch the app locally and create a new report.
2. Attempt to upload the file located at `audit/fixtures/workbooks/should_fail\audit_findings\has_duplicate_finding_reference_with_different_finding_details` and observe the error message.
3. Update the workbook by either:
   - Changing the reference number to a different value, or
   - Modifying the findings details in row 3 to match the details in row 2.
4. Try uploading the updated workbook again and confirm that it uploads successfully.

![image](https://github.com/user-attachments/assets/1c4c310b-221b-41d4-8fdc-db7a7b301c47)


## PR Checklist: Submitter


## PR Checklist: Reviewer


The larger the PR, the stricter we should be about these points.

## Pre Merge Checklist: Merger
